mmv1/templates/terraform/custom_delete/storage_folder_delete.go.tmpl

```diff
@@ -54,14 +54,33 @@ for deleteObjectError == nil {
     wp.StopWait()
 }
 
-if listError != nil {
+err = retry.Retry(1*time.Minute, func() *retry.RetryError {
+    err := config.NewStorageClient(userAgent).Folders.Delete(bucket, name).Do()
+    if err == nil {
+        return nil
+    }
+    if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 429 {
+        return retry.RetryableError(gerr)
+    }
+    return retry.NonRetryableError(err)
+})
+
+if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 409 && strings.Contains(gerr.Message, "not empty") && listError != nil {
     return fmt.Errorf("could not delete non-empty folder due to error when listing contents: %v", listError)
 }
-if deleteObjectError != nil {
+if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 409 && strings.Contains(gerr.Message, "not empty") && deleteObjectError != nil {
     return fmt.Errorf("could not delete non-empty folder due to error when deleting contents: %v", deleteObjectError)
 }
+if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 409 && strings.Contains(gerr.Message, "not empty") && !d.Get("force_destroy").(bool) {
+    return fmt.Errorf("Sub folders or items may exist within folder, use force_destroy to true to delete all subfolders: %v", err)
+}
+
+if err == nil {
```

Comment:
Nit: we might want to still log the err if it is not nil and we get to this point. While it wouldn't change behavior, it might help when debugging, since the error is otherwise swallowed.

Reply:
added log for the err